### PR TITLE
MGMT-3397 Do not apply 3->2->3 patch from Openshift 4.7 and later

### PR DESCRIPTION
**Pull request by ori-amizur**
/cc @eranco74 
/cc @tsorya 
/cc @romfreiman

### CODE REVIEW

File: src/installer/installer_test.go [hunk 1]
```diff
@@ -238,62 +238,75 @@ var _ = Describe("installer HostRoleMaster role", func() {
...
 			daemonReload(nil)
 		}
-
-		It("bootstrap role happy flow", func() {
-			updateProgressSuccess([][]string{{string(models.HostStageStartingInstallation), conf.Role},
-				{string(models.HostStageWaitingForControlPlane)},
-				{string(models.HostStageInstalling), string(models.HostRoleMaster)},
-				{string(models.HostStageWritingImageToDisk)},
-				{string(models.HostStageRebooting)},
+		for _, version := range []string{"4.6", "4.7"} {
```
Comment (romfreiman) [inline src/installer/installer_test.go:241]: 4.7.1? 4.8?

File: src/installer/installer.go [hunk 2]
```diff
@@ -376,39 +376,55 @@ func (i *installer) waitForControlPlane(ctx context.Context, kc k8s_client.K8SCl
...
-func (i *installer) waitForMinMasterNodes(ctx context.Context, kc k8s_client.K8SClient) error {
-	if err := i.waitForNetworkType(kc); err != nil {
+func (i *installer) shouldControlPlaneReplicasPatchApplied(kc k8s_client.K8SClient) (bool, error) {
+	controlPlanePatchRequired, err := utils.IsVersionLessThan47(i.Config.OpenshiftVersion)
+	if err != nil {
+		i.log.WithError(err).Errorf("Failed to get compare OCP version")
+		return false, err
+	}
+	if !controlPlanePatchRequired {
+		return false, nil
```
Comment (romfreiman) [inline src/installer/installer.go:386]: please log

Comment (ori-amizur) [reply src/installer/installer.go:387]: OK

File: src/installer/installer_test.go (hunk 1 above)
Comment (ori-amizur) [reply src/installer/installer_test.go]: OK

**Pull request merged**